security.py

```diff
@@ -4,24 +4,51 @@
 
 # We should be importing Fernet to be used here, not through the env file
 
-def authenticate(data, key, index):
+DISPATCHER = {
+    "CAMERAS" : env.CAMERAS,
+    "RECIPIENTS" : env.RECIPIENTS,
+    "CREDENTIALS" : env.CREDENTIALS,
+    "CARRIERS" : env.CARRIERS
+}
+
+def authenticate(username, password):
     """
     Takes an input value and verifies against encrypted value.
     """
+    dec_user = __decrypt("CREDENTIALS", "user_login", "username")
+    dec_pass = __decrypt("CREDENTIALS", "user_login", "password")
 
-    # WE DONT NEED TO ENCRYPT ANYMORE SINCE FERNET IS NOT A "DETERMINISTIC AUTHENTICATED CYPHER". THIS 
-    # ENCRYPTED PASSWORD IS DIFFERENT THAN VAULT ENCRYPTED PASSWORD
-    # enc_data = __encrypt(data, False)
+    if dec_user == -1 or dec_pass == -1 or dec_user != username or dec_pass != password:
+        raise Exception("Failed to Authenticate Username or Password")
 
-    # if (key in env.CREDENTIALS.keys()):
-    #     value = __get_value(env.CREDENTIALS, key, index)
+def get_value(env_var, key, field):
+    """
+    Reads from vault and returns decrypted value.
+    """
+    dec_val = ""
+    try:
+        dec_val = __decrypt(env_var, key, field)
+    except Exception as e:
+        raise e
     
-    dec_pwd = __decrypt(key, index)
+    return dec_val
 
-    if dec_pwd == -1 or dec_pwd != data:
-        raise Exception("Failed to Authenticate Username or Password")
+def get_keys(env_var, key=-1):
+    """
+    Reads from vault and returns env keys.
+    """
+    env_var = env_var.upper()
+    try:
+        if key == -1:
+            # Return outer keys
+            return DISPATCHER.get(env_var).keys()
+        # Return inner keys
+        return DISPATCHER.get(env_var).get(key).keys()
+
+    except:
+        raise Exception("Invalid environment variable or key name")
 
-def __encrypt(val, write, key=-1, index=-1):
+def __encrypt(env_var, val, write, key, field):
```

Comment:
Maybe we should add a default value to 'write'. Or maybe we should remove it in general since it isn't used anymore :D

Reply:
Since it is out of scope of this PR, we will revisit this!